File: wk2_assignments/assignment_1/node/Grader.js
class Grader {
    grade = function(gradeInput){
        const gradeNum = parseInt(gradeInput);
        this.gradeLetter = "";
        this.feedback = "";
        
        if(gradeNum >= 90){
            this.gradeLetter = "A";
            this.feedback = "You completed the requirements for the assignment, Great Job!";
            
        }
        else if(gradeNum >= 80){
            this.gradeLetter = "B";
            this.feedback = "You have a few small errors in your code.";
        }
        else if(gradeNum >= 70){
            this.gradeLetter = "C";
            this.feedback = "You are missing a requirement(s) for the assignment";
        }
        else if(gradeNum >= 60){
            this.gradeLetter = "D";
            this.feedback = "You are missing some requirements(s) for the assignment, and have errors.";
        }
        else{
            this.gradeLetter = "F";
            this.feedback = "You did not fulfill the requirements for the assignment! Try Again.";
        }
        
        return [this.gradeLetter, this.feedback];
    }
}

module.exports = Grader;

File: wk2_assignments/assignment_1/python/Grader.py
class Grader:
        def lettergrade(self, gradeInput):
            gradeNum = float(gradeInput)
            gradeLetter = ""
            feedback = ""
            if gradeNum >= 90:
                gradeLetter = "A"
                feedback = "You completed the requirements for the assignment, Great Job!"
            elif gradeNum >= 80:
                gradeLetter = "B"
                feedback = "You have a few small errors in your code."
            elif gradeNum >= 70:
                gradeLetter = "C"
                feedback = "You are missing a requirement(s) for the assignment"
            elif gradeNum >= 60:
                gradeLetter = "D"
                feedback = "You are missing some requirements(s) for the assignment, and have errors."
            else:
                gradeLetter = "F"
                feedback = "You did not fulfill the requirements for the assignment! Try Again."
            
            gradeObj = {"gradeLetter": gradeLetter, "feedback": feedback}
            return gradeObj
